starboard/shared/starboard/player/player_worker.cc

```diff
@@ -199,15 +202,19 @@ void PlayerWorker::DoInit() {
   Handler::UpdatePlayerErrorCB update_player_error_cb;
   update_player_error_cb =
       std::bind(&PlayerWorker::UpdatePlayerError, this, _1, _2);
-  if (handler_->Init(
-          player_, std::bind(&PlayerWorker::UpdateMediaInfo, this, _1, _2, _3),
-          std::bind(&PlayerWorker::player_state, this),
-          std::bind(&PlayerWorker::UpdatePlayerState, this, _1),
-          update_player_error_cb)) {
+  HandlerResult result = handler_->Init(
+      player_, std::bind(&PlayerWorker::UpdateMediaInfo, this, _1, _2, _3),
+      std::bind(&PlayerWorker::player_state, this),
+      std::bind(&PlayerWorker::UpdatePlayerState, this, _1),
+      update_player_error_cb);
+  if (result.success) {
     UpdatePlayerState(kSbPlayerStateInitialized);
   } else {
-    UpdatePlayerError(kSbPlayerErrorDecode,
-                      "Failed to initialize PlayerWorker with unknown error.");
+    std::string error_message = "Failed to initialize PlayerWorker.";
+    if (!result.error_message.empty()) {
```

Comment:
This code is repeated quite a few times in the file, consider abstracting them into `UpdatePlayerError()`.